src/xml_parser_main_metadata_together.c

```diff
@@ -59,23 +60,12 @@ call_user_callback_if_package_finished(cr_Package *pkg, cr_CbData *cb_data, GErr
         cb_data->in_progress_count_primary--;
         cb_data->in_progress_count_filelists--;
         cb_data->in_progress_count_other--;
-
-        // call user package callback
-        GError *tmp_err = NULL;
-        if (cb_data->pkgcb && cb_data->pkgcb(pkg, cb_data->pkgcb_data, &tmp_err)) {
-            if (tmp_err)
-                g_propagate_prefixed_error(err, tmp_err, "Parsing interrupted: ");
-            else
-                g_set_error(err, ERR_DOMAIN, CRE_CBINTERRUPTED, "Parsing interrupted");
-            return CR_CB_RET_ERR;
-        } else {
-            // If callback return CRE_OK but it simultaneously set
-            // the tmp_err then it's a programming error.
-            assert(tmp_err == NULL);
-        };
+        cb_data->finished_pkgs_list = g_slist_append(cb_data->finished_pkgs_list, pkg);
```

Comment:
One minor concern I have is that we're appending to the end of a linked list without using a tail pointer, so it has to iterate the list every time.  It's not very efficient.  I'd like to address that somehow before merging.

Reply:
Perhaps a queue ought to be used instead